crates/lib3h/src/rrdht_util/location.rs

```diff
@@ -0,0 +1,119 @@
+use std::num::Wrapping;
+
+/// An rrdht "location" refers to a hash of a signing public key
+/// compressed down into 32 unsigned bits (u32)
+/// We use "Wrapping" math because location offsets or "arcs" wrap...
+/// i.e. are mapped onto a circle
+#[derive(Shrinkwrap, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[shrinkwrap(mutable)]
+pub struct Location(pub Wrapping<u32>);
+
+impl Location {
+    /// get the distance from this location to an `other` location
+    /// this distance is processed forward taking wrapping into account
+    #[allow(clippy::trivially_copy_pass_by_ref)]
+    pub fn forward_distance_to(&self, other: Location) -> u64 {
```

Comment:
I wonder, do we not need the symmetric distance, i.e. distance taking both directions around the circle into account?

Reply:
I don't have a current mathematical need for that, but it should be easy to add if we come across one : )